packages/app/src/components/authorize/AuthorizeToken.tsx

```diff
@@ -0,0 +1,50 @@
+import { FC, MouseEventHandler } from "react";
+import { Alert, Tooltip, OverlayTrigger } from "react-bootstrap";
+import { constants } from "ethers";
+import { addresses } from "@tender/contracts";
+import { BigNumber } from "@ethersproject/bignumber";
+
+type Props = {
+  protocolName: string;
+  tokenSymbol: string;
+  isTokenAuthorized: boolean;
+  approveToken: (contract: string, amount: BigNumber) => Promise<void>;
+};
+
+const AuthorizeToken: FC<Props> = ({ tokenSymbol, protocolName, approveToken, isTokenAuthorized }) => {
+  const handlePressTrade: MouseEventHandler<HTMLDivElement> = async (e) => {
+    e.preventDefault();
+
+    if (!isTokenAuthorized) {
+      await approveToken(addresses[protocolName].swap, constants.MaxUint256);
+    }
+  };
+
+  return (
+    <Alert
```

Comment:
This needs to be a button not an alert , not sure why we were thinking an alert would be a good fit for a CTA

Reply:
right, at the beginning I wasn't sure if it will be just an alert, and the action will be tied to the button under it, and after just kept it like that

Reply:
THis is how Uniswap does it
![image](https://user-images.githubusercontent.com/22256858/124383285-f7df2400-dccb-11eb-9e16-feb6f76a36be.png)


It's a button with a tooltip inside of it